test/behaviour/reasoner/verification/ForwardChainingMaterialiser.java

```diff
@@ -110,10 +116,58 @@ Optional<Materialisation> conditionMaterialisations(com.vaticle.typedb.core.logi
         return materialisations.forCondition(rules.get(rule), conditionAnswer);
     }
 
+    private List<Set<Rule>> getStratifiedRules() {
```

Comment:
fyi we don't really use the logic terminology like this very often - only if we can't find a name that requires less context/understanding for future readers. For example, the validation logic in TypeDB does stratification checking without using the word stratification...

Reply:
That's true, but it would be even easier if you can get away without googling by naming it something like "ruleCyclesWithoutNegations" or something, that would be more self-explanatory

not sure exactly that stratification is anymore after i understood it once and sidestepped needing the term in core since then :)

Reply:
which illustrates my point actually

Reply:
well if you're using `strata` you can keep the naming using `strata` consistent and don't need the long name... i have totally forgotten stratification so i will have to think about this separately! feel free to proceed as it was